Spokestack/SpeechPipeline.swift

```diff
@@ -7,184 +7,195 @@
     }
     
     /// MARK: Pipeline control
     
     /**
      Activates speech recognition. The pipeline remains active until the user stops talking or the activation timeout is reached.
- 
+     
      Activations have configurable minimum/maximum lengths. The minimum length prevents the activation from being aborted if the user pauses after saying the wakeword (which deactivates the VAD). The maximum activation length allows the activation to timeout if the user doesn't say anything after saying the wakeword.
-    
-     The wakeword detector can be used in a multi-turn dialogue system. In such an environment, the user is not expected to say the wakeword during each turn. Therefore, an application can manually activate the pipeline by calling `activate` (after a system turn), and the wakeword detector will apply its minimum/maximum activation lengths to control the duration of the activation.
      
-     - SeeAlso: `wakeActiveMin`, `wakeActiveMax`
+     The wakeword detector can be used in a multi-turn dialogue system. In such an environment, the user is not expected to say the wakeword during each turn. Therefore, an application can manually activate the pipeline by calling `activate` (after a system turn), and the wakeword detector will apply its minimum/maximum activation lengths to control the duration of the activation.
     */
     @objc public func activate() -> Void {
-        self.wakewordRecognizerService.stopStreaming(context: self.context)
-        self.speechRecognizerService.startStreaming(context: self.context)
+        self.context.isActive = true
+        self.context.listeners.forEach({ listener in
+            listener.didActivate()
+        })
     }
     
     /// Deactivates speech recognition.  The pipeline returns to awaiting either wakeword activation or an explicit `activate` call.
     /// - SeeAlso: `activate`
     @objc public func deactivate() -> Void {
-        self.speechRecognizerService.stopStreaming(context: self.context)
-        self.wakewordRecognizerService.startStreaming(context: self.context)
+        self.context.isActive = false
+        self.context.listeners.forEach({ listener in
+            listener.didDeactivate()
+        })
     }
     
     /// Starts  the speech pipeline.
     ///
-    /// The pipeline starts in a deactivated state, awaiting either a wakeword activation or an explicit call to `activate`.
+    /// The pipeline starts in a deactivated state, awaiting either a triggered activation  from a wakeword or VAD, or an explicit call to `activate`.
     @objc public func start() -> Void {
-        if (self.context.isActive) {
-            self.stop()
-        }
-        AudioController.sharedInstance.startStreaming(context: self.context)
-        self.wakewordRecognizerService.startStreaming(context: self.context)
-        self.speechConfiguration?.delegateDispatchQueue.async {
-            self.pipelineDelegate?.didStart()
+        
```

Comment:
Update: Android just got an adjustment in this area. You'll still want to avoid the list of stages growing without bound, but calling start on a running pipeline isn't necessarily an error condition. See https://github.com/spokestack/spokestack-android/pull/81